### ISSUE TITLE
Update to Webpack 4 and use @babel/core to take advantage of modern features

### ISSUE THREAD

**Issue by KenEucker** [ahmadawais/create-guten-block#163]
## Feature Request
Update to webpack 4 and use @babel/core

**Is your feature request related to a problem? Please describe.**
babel-core hasn't been updated in a year, and newer webpack plugins don't work. Updating the project to use @babel/core and webpack 4 will allow developrs to use modern features in their code.

The issue I am running into is that newer plugins needed for the React code I am using in my project appear to not be compatible with the webpack and babel versions used in cgb. This has forced me to eject my project and attempt to support these newer features in the custom build process. However, in attempting to do so, I've run into a litany of issues with other, outdated, packages.

**Describe the solution you'd like**
The issues I am encountering are not obscure or edge case. I feel that things are just a little out of date here and could benefit from being updated.

**Describe alternatives you've considered**
I've ejected my project and modified the ejected scripts to use the updated babel features found in my project.

**Teachability, Documentation, Adoption, Migration Strategy**
I don't know what the ramifications would be on legacy babel code when moving from 6-7, but if there are any then the documentation would need to call this out in the changelog.

**Comment by ahmadawais** [comment]
Feel free to contribute I will happily accept this upgrade. Sadly, personally, haven't been able to find companies/individuals willing to pay for my bills while I work through this myself.

**Comment by KenEucker** [comment]
@ahmadawais Sorry to hear that! I know how that goes as I've been trying to do the work for free thing for a couple of years.

I noticed, and I apologize because I noticed after creating this issue, that there is PR for updating webpack:

https://github.com/ahmadawais/create-guten-block/pull/158

I am going to start there and also see if I can get @babel/core in there as well. If I do get a functioning update working I will create a PR.

Cheers!

**Comment by KenEucker** [comment]
@ahmadawais 

I've forked the up to date master and have created a branch 'webpack4@babelcore' to demonstrate the changes I'm proposing and have tested on my own project. Is there more testing I can do ahead of the canary branch?

I'm following the workflow steps outlined in the contributing guideline. I believe the next step is for me to request publishing access to all the npm packages in CGB. Following that, it is my understanding that I next run the command `npm run releaseCanary` and can begin canary testing. Is that correct?

Happy to contribute,

**Comment by ahmadawais** [comment]
Yes, you are correct. But before we do that, can you open a PR for that and allow me or @asharirfan to release canary for you? That would help. Once your first PR is in, I'd love to have you on the team improving create-guten-block. 

Your contributions are super welcomed!
